src/protocol.cpp

```diff
@@ -46,46 +46,44 @@ const char *CFCHECKPT="cfcheckpt";
 const char *WTXIDRELAY="wtxidrelay";
 } // namespace NetMsgType
 
-/** All known message types. Keep this in the same order as the list of
+/** All known message types including the short-ID. Keep this in the same order as the list of
  * messages above and in protocol.h.
  */
-const static std::string allNetMessageTypes[] = {
-    NetMsgType::VERSION,
-    NetMsgType::VERACK,
-    NetMsgType::ADDR,
-    NetMsgType::ADDRV2,
-    NetMsgType::SENDADDRV2,
-    NetMsgType::INV,
-    NetMsgType::GETDATA,
-    NetMsgType::MERKLEBLOCK,
-    NetMsgType::GETBLOCKS,
-    NetMsgType::GETHEADERS,
-    NetMsgType::TX,
-    NetMsgType::HEADERS,
-    NetMsgType::BLOCK,
-    NetMsgType::GETADDR,
-    NetMsgType::MEMPOOL,
-    NetMsgType::PING,
-    NetMsgType::PONG,
-    NetMsgType::NOTFOUND,
-    NetMsgType::FILTERLOAD,
-    NetMsgType::FILTERADD,
-    NetMsgType::FILTERCLEAR,
-    NetMsgType::SENDHEADERS,
-    NetMsgType::FEEFILTER,
-    NetMsgType::SENDCMPCT,
-    NetMsgType::CMPCTBLOCK,
-    NetMsgType::GETBLOCKTXN,
-    NetMsgType::BLOCKTXN,
-    NetMsgType::GETCFILTERS,
-    NetMsgType::CFILTER,
-    NetMsgType::GETCFHEADERS,
-    NetMsgType::CFHEADERS,
-    NetMsgType::GETCFCHECKPT,
-    NetMsgType::CFCHECKPT,
-    NetMsgType::WTXIDRELAY,
-};
-const static std::vector<std::string> allNetMessageTypesVec(std::begin(allNetMessageTypes), std::end(allNetMessageTypes));
+const static std::map<uint8_t, std::string> allNetMessageTypes = {
+    {37, NetMsgType::VERSION},
+    {36, NetMsgType::VERACK},
+    {13, NetMsgType::ADDR},
+    {45, NetMsgType::ADDRV2},
+    {46, NetMsgType::SENDADDRV2},
+    {27, NetMsgType::INV},
+    {24, NetMsgType::GETDATA},
+    {29, NetMsgType::MERKLEBLOCK},
+    {22, NetMsgType::GETBLOCKS},
+    {25, NetMsgType::GETHEADERS},
+    {35, NetMsgType::TX},
+    {26, NetMsgType::HEADERS},
+    {14, NetMsgType::BLOCK},
+    {21, NetMsgType::GETADDR},
+    {28, NetMsgType::MEMPOOL},
+    {31, NetMsgType::PING},
+    {32, NetMsgType::PONG},
+    {30, NetMsgType::NOTFOUND},
+    {20, NetMsgType::FILTERLOAD},
+    {18, NetMsgType::FILTERADD},
+    {19, NetMsgType::FILTERCLEAR},
+    {34, NetMsgType::SENDHEADERS},
+    {17, NetMsgType::FEEFILTER},
+    {33, NetMsgType::SENDCMPCT},
+    {16, NetMsgType::CMPCTBLOCK},
+    {23, NetMsgType::GETBLOCKTXN},
+    {15, NetMsgType::BLOCKTXN},
+    {38, NetMsgType::GETCFILTERS},
+    {39, NetMsgType::CFILTER},
+    {40, NetMsgType::GETCFHEADERS},
+    {41, NetMsgType::CFHEADERS},
+    {42, NetMsgType::GETCFCHECKPT},
+    {43, NetMsgType::CFCHECKPT},
+    {44, NetMsgType::WTXIDRELAY}};
```

Comment:
Checked that this matches the table in https://gist.github.com/dhruv/5b1275751bc98f3b64bcafce7876b489#message-type-id  (but not the one in Jonas' PR https://github.com/bitcoin/bips/blob/2d331e8a897abdd09fc3af8e352bb0b74f2cbc93/bip-0324.mediawiki#message-type-id )